src/features/generic_events/aggregate/user-actions/user-actions-aggregator.js

```diff
@@ -28,40 +42,61 @@ export class UserActionsAggregator {
    */
   process (evt, targetFields) {
     if (!evt) return
-    const { selectorPath, nearestTargetFields } = getSelectorPath(evt, targetFields)
-    const aggregationKey = getAggregationKey(evt, selectorPath)
+    const targetElem = OBSERVED_WINDOW_EVENTS.includes(evt.type) ? window : evt.target
+    const selectorInfo = analyzeElemPath(targetElem, targetFields)
+
+    // if selectorInfo.path is undefined, aggregation will be skipped for this event
+    const aggregationKey = getAggregationKey(evt, selectorInfo.path)
     if (!!aggregationKey && aggregationKey === this.#aggregationKey) {
       // an aggregation exists already, so lets just continue to increment
       this.#aggregationEvent.aggregate(evt)
     } else {
       // return the prev existing one (if there is one)
       const finishedEvent = this.#aggregationEvent
-      // then set as this new event aggregation
+      this.#ufEnabled && this.#deadClickCleanup()
+
+      // then start new event aggregation
       this.#aggregationKey = aggregationKey
-      this.#aggregationEvent = new AggregatedUserAction(evt, selectorPath, nearestTargetFields)
+      this.#aggregationEvent = new AggregatedUserAction(evt, selectorInfo)
+      if (this.#ufEnabled && evt.type === 'click' && (selectorInfo.hasButton || selectorInfo.hasLink)) {
+        this.#deadClickSetup(this.#aggregationEvent)
+      }
       return finishedEvent
     }
   }
-}
 
-/**
- * Generates a selector path for the event, starting with simple cases like window or document and getting more complex for dom-tree traversals as needed.
- * Will return a random selector path value if no other path can be determined, to force the aggregator to skip aggregation for this event.
- * @param {Event} evt
- * @returns {string}
- */
-function getSelectorPath (evt, targetFields) {
-  let selectorPath; let nearestTargetFields = {}
-  if (OBSERVED_WINDOW_EVENTS.includes(evt.type) || evt.target === window) selectorPath = 'window'
-  else if (evt.target === document) selectorPath = 'document'
-  // if still no selectorPath, generate one from target tree that includes elem ids
-  else {
-    const { path, nearestFields } = generateSelectorPath(evt.target, targetFields)
-    selectorPath = path
-    nearestTargetFields = nearestFields
+  #deadClickSetup (userAction) {
+    if (this.#startObserver()) {
```

Comment:
This is convoluted. The short circuit logic is double nested & not obvious. I suggest you check for the falsey case right here and let startObserver simply do as its name implies rather than actually be "start observer but only if there isn't a dead click timer")